### PR TITLE
Contract Updates & Client Integration

### LINKED ISSUE #287: Audit fixes 1-4
SigP report for `ce7f958` (`/contracts` unchanged to tag: `v0.3.0`).

### PR DESCRIPTION
**Pull request by jacque006**
## What is this PR doing?
Continuation of https://github.com/web3well/bls-wallet/pull/350
- Contract audit fixes 1-4
- BLSWalletWrapper updates to handle changing addresses referred to by a key.

## How can these changes be manually tested?
- Use the Quill extension w/ a dApp
- All other changes tested by CI/GH actions

## Does this PR resolve or contribute to any issues?
Resolves https://github.com/web3well/bls-wallet/issues/287

## Checklist
- [x] I have manually tested these changes
- [x] Post a link to the PR in the group chat

## Guidelines
- If your PR is not ready, mark it as a draft
- The `resolve conversation` button is for reviewers, not authors
  - (But add a 'done' comment or similar)

## TODO
- [x] Merge in changes from https://github.com/web3well/bls-wallet/pull/340
- [x] Add test case for `VG disallows calling transferOwnership`
- [x] Add test case for `VG disallows calling renounceOwnership`
- [x] Add test case for `VG: first param to proxy admin is not calling wallet`
- [x] Update & publish experimental client and pin in consuming components.
- [x] Regression test extension, which will also test aggregator.
- [x] Fix aggregator type issue (https://github.com/web3well/bls-wallet/issues/396)

## Post Merge
- [ ] Tag new repo release
- [x] Publish and pin in repo `bls-wallet-clients@0.8.0`
- [x] Deploy contract updates to Goerli Arbitrum
- [ ] Deploy updated aggregators
- [ ] Update other projects/repos?

### CODE REVIEW

File: contracts/test/upgrade-test.ts [hunk 1]
```diff
@@ -347,4 +364,54 @@ describe("Upgrade", async function () {
...
+      wallet,
+      "renounceOwnership",
+      [],
+    );
+    const opResults = getOperationResults(txnReceipt);
+
+    expect(opResults).to.have.lengthOf(1);
+    expect(opResults[0].error.actionIndex.toNumber()).to.eql(0);
+    expect(opResults[0].error.message).to.eql("VG: cannot change ownership");
+  });
```
Comment (jacque006) [inline contracts/test/upgrade-test.ts]: Need to investigate and fix this

Comment (voltrevo) [reply contracts/test/upgrade-test.ts]: Fixed: https://github.com/web3well/bls-wallet/pull/391

File: .github/workflows/aggregator.yml [hunk 2]
```diff
@@ -6,13 +6,19 @@ on:
...
-      - 'main'
     paths:
       - 'aggregator/**'
+      # Check for breaking changes from contracts
+      - 'contracts/**'
       - '.github/workflows/aggregator.yml'
+    branches-ignore:
+      # Changes targeting this branch should be tested+fixed when being merged
+      # into main
+      - contract-updates
```
Comment (jacque006) [inline .github/workflows/aggregator.yml:21]: Was this disabled as the agg tests were failing/taking too long? We should run all relevant tests on any PR so the author knows if they are breaking/not breaking something.

Comment (jacque006) [reply .github/workflows/aggregator.yml:21]: Discussed with Andrew, this makes it so contracts changes that break agg are ok going into `contract-updates` but not into main.

**Review by voltrevo — APPROVED** [review]

**Pull request merged**